manifest: Update Matter SDK revision to pull new crypto changes (#13540)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst

      * The ``CONFIG_CHIP_SED_ACTIVE_INTERVAL`` Kconfig option was renamed to :kconfig:option:`CONFIG_CHIP_ICD_FAST_POLLING_INTERVAL`.
      * The ``CONFIG_CHIP_SED_ACTIVE_THRESHOLD`` Kconfig option was renamed to :kconfig:option:`CONFIG_CHIP_ICD_ACTIVE_MODE_THRESHOLD`.

* For Matter over Thread samples, starting from this release, the cryptography backend enabled by default is PSA Crypto API instead of Mbed TLS.
  Be aware of the change and consider the following when migrating to |NCS| v2.6.0:

  * You can keep using Mbed TLS API as the cryptography backend by disabling PSA Crypto API.
    You can disable it by setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n``.

>   * * Thread libraries will be built with PSA Crypto API enabled without Mbed TLS support.

Reviewer: ```suggestion
  * Thread libraries will be built with PSA Crypto API enabled without Mbed TLS support.
```

    This means that if you set the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n``, you must also build the Thread libraries from sources.

    To :ref:`inherit Thread certification <ug_matter_device_certification_reqs_dependent>` from Nordic Semiconductor, you must use the PSA Crypto API backend.
  * The device will automatically migrate all operational keys from the Mbed TLS database to the PSA ITS secure storage.

>     This means that all keys needed to establish the secure connection between Matter nodes will be moved to the PSA ITS secure storage.

Reviewer: ```suggestion
    This means that all keys needed to establish the secure connection between Matter nodes will be moved to the PSA ITS secure storage.

```

    If the migration fails, the ``kOperationalKeystoreMigrationFailed`` public event will be emitted and dispatched in the Matter event handler.
    The default reaction to migration failure is a factory reset of the device.

>     To change the default reaction, set the :kconfig:option:`NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE` Kconfig option to ``n`` and implement the reaction in your Matter event handler.

Reviewer 2: Add `CONFIG` prefix to the kconfig name?

Reviewer: Good catch.

>     To change the default reaction, set the :kconfig:option:`NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE` Kconfig option to ``n`` and implement the reaction in your Matter event handler.

Reviewer 2: I'm wondering about this one. It assumes the customer uses our common event module and sourced our Kconfig file from the common directory, which is actually true only for our samples and not the customer's application. 

I was not involved in adding PSA support, so ignore it if I'm not right, but would it make sense to catch an event and perform factory reset somewhere in the Matter core/nRF platform instead of application?

One way or another I think it should be mentioned that some additional files has to be sourced to support the mentioned Kconfig and method.

Author: In Matter Core the platform-specific code currently, we don't have such an event handler to catch that event. We register the handler like that one during Server Init and use it in our common directory so for now I use it to implement the raction to migration failure. I agree that we can create an additional event handler in the platform/nrfconnect and catch that event there, but we I think we should discuss it first and do it as a follow-up to this task.

Reviewer 2: Ok, let's skip moving the handler then. And how about sourcing Kconfig? Could we add sentence explaining it should be added? I mean, for me now it sounds, like everything should work automatically - the migration, pushing and handling the event. Actually it's not true, because without sourcing common Kconfig in the app user will not get it.

      * The ``CONFIG_CHIP_SED_ACTIVE_INTERVAL`` Kconfig option was renamed to :kconfig:option:`CONFIG_CHIP_ICD_FAST_POLLING_INTERVAL`.
      * The ``CONFIG_CHIP_SED_ACTIVE_THRESHOLD`` Kconfig option was renamed to :kconfig:option:`CONFIG_CHIP_ICD_ACTIVE_MODE_THRESHOLD`.

* For Matter over Thread samples, starting from this release, the cryptography backend enabled by default is PSA Crypto API instead of Mbed TLS.
  Be aware of the change and consider the following when migrating to |NCS| v2.6.0:

  * You can keep using Mbed TLS API as the cryptography backend by disabling PSA Crypto API.
    You can disable it by setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n``.
  * Thread libraries will be built with PSA Crypto API enabled without Mbed TLS support.
    This means that if you set the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n``, you must also build the Thread libraries from sources.

    To :ref:`inherit Thread certification <ug_matter_device_certification_reqs_dependent>` from Nordic Semiconductor, you must use the PSA Crypto API backend.

>   * The device will automatically migrate all operational keys from the Mbed TLS database to the PSA ITS secure storage.

Reviewer 3: There's no such thing as Mbed TLS. How about the Matter's generic persistent storage?

  * You can keep using mbedTLS API as the cryptography backend by disabling PSA Crypto API.
    You can disable it by setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n``.
  * Thread libraries will be built with PSA Crypto API enabled without mbedTLS support.

>     It means after setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n`` Thread libraries you must build from sources.

Reviewer 4: ```suggestion
    It means after setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n`` you must build Thread libraries from sources.
```
Although I love master Yoda style...

>     Moreover, to :ref:`inherit Thread certification <ug_matter_device_certification_reqs_dependent>` from Nordic Semiconductor you MUST use the PSA Crypto API backend.

Reviewer: ```suggestion

    To :ref:`inherit Thread certification <ug_matter_device_certification_reqs_dependent>` from Nordic Semiconductor, you must use the PSA Crypto API backend.
 
```

>   * The device will automatically migrate all operational keys from the mbedTLS database to PSA ITS secure storage.

Reviewer: ```suggestion
  * The device will automatically migrate all operational keys from the Mbed TLS database to the PSA ITS secure storage.
```

>     It means that all keys needed to establish the secure connection between Matter nodes will be moved to the PSA ITS secure storage.

Reviewer: ```suggestion
    This means that all keys needed to establish the secure connection between Matter nodes will be moved to the PSA ITS secure storage.
 
```

>     If the migration fails the ``kOperationalKeystoreMigrationFailed`` Matter's public event will be emitted and dispatched in the ``Matter event handler``.

Reviewer: ```suggestion
    If the migration fails, the ``kOperationalKeystoreMigrationFailed`` public event will be emitted and dispatched in the Matter event handler.
```
I don't think the event handler is supposed to have code formatting.

    The default reaction to migration failure is a factory reset of the device.

>     To change the default reaction set the :kconfig:option:`NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE` Kconfig option to `n` and implement the reaction in your ``Matter event handler``.

Reviewer: ```suggestion
    To change the default reaction, set the :kconfig:option:`NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE` Kconfig option to ``n`` and implement the reaction in your Matter event handler.
```

>     For more information about the ``Matter event handler`` see the :file:`matter_event_handler.h` file.

Reviewer: ```suggestion
    For more information about the Matter event handler, see the :file:`matter_event_handler.h` header file.

```
